library/core/src/ptr/mod.rs

```diff
@@ -2021,54 +2026,68 @@ pub const unsafe fn write_unaligned<T>(dst: *mut T, src: T) {
     }
 }
 
-/// Performs a volatile read of the value from `src` without moving it. This
-/// leaves the memory in `src` unchanged.
-///
-/// Volatile operations are intended to act on I/O memory, and are guaranteed
-/// to not be elided or reordered by the compiler across other volatile
-/// operations.
-///
-/// # Notes
-///
-/// Rust does not currently have a rigorously and formally defined memory model,
-/// so the precise semantics of what "volatile" means here is subject to change
-/// over time. That being said, the semantics will almost always end up pretty
-/// similar to [C11's definition of volatile][c11].
-///
-/// The compiler shouldn't change the relative order or number of volatile
-/// memory operations. However, volatile memory operations on zero-sized types
-/// (e.g., if a zero-sized type is passed to `read_volatile`) are noops
-/// and may be ignored.
+/// Performs a volatile read of the value from `src` without moving it.
+///
+/// Rust does not currently have a rigorously and formally defined memory model, so the precise
+/// semantics of what "volatile" means here is subject to change over time. That being said, the
+/// semantics will almost always end up pretty similar to [C11's definition of volatile][c11].
```

Comment:
I wonder if we still need this paragraph. The next paragraph says basically as much as C says, after all.

@rust-lang/opsem what do you think?

Reply:
If nobody else has an opinion, I'd say please remove this paragraph.